src/mappings/position-registry.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  OwnershipTransferred,
+  PositionDeployed as PositionDeployedEvent,
+  StrategyAdded as StrategyAddedEvent,
+  PositionRegistry as PositionRegistryContract,
+  IsActiveUpdated,
+  FeeCollectorUpdated,
+  FeePolicyUpdated,
+  ImplementationUpdated,
+} from '../../generated/PositionRegistry/PositionRegistry'
+import { PositionRegistry, Position, Strategy } from '../../generated/schema'
+import { Position as PositionTemplate } from '../../generated/templates'
+import { ADDRESS_ZERO, BIGINT_ONE, BIGINT_ZERO } from '../utils/constants'
+import { loadOrCreateSmartAccount } from './entry-point'
+
+export function handlePositionDeployed(event: PositionDeployedEvent): void {
+  // at this point we know that positionRegistry is initialized
+  const registry = PositionRegistry.load(event.address)!
+
+  // create smartAccount entity if smartAccount is deployed before startBlock.
+  // load smartAccount entity, Set PositionRegistry, this update mark smartAccount as Odyssey smartAccount.
+  const smartAccount = loadOrCreateSmartAccount(event.params.owner)
+  if (smartAccount.positionRegistry == ADDRESS_ZERO) {
+    smartAccount.positionRegistry = registry.id
+    registry.smartAccountCount = registry.smartAccountCount.plus(BIGINT_ONE)
+  }
+
+  // create new position entity
+  const position = new Position(event.params.position)
+  position.owner = smartAccount.id
+  position.strategyId = event.params.strategyId
+  position.createdAt = event.block.timestamp
+  position.openedAt = BIGINT_ZERO
+  position.txCount = BIGINT_ZERO
+  position.totalAllocated = BIGINT_ZERO
+  position.pricePerShare = BIGINT_ZERO
+  position.asset = ADDRESS_ZERO
+  position.isOutdated = false
+
+  smartAccount.positionCount = smartAccount.positionCount.plus(BIGINT_ONE)
+
+  registry.positionCount = registry.positionCount.plus(BIGINT_ONE)
+
+  position.save()
+  // create new datasource for the position
+  PositionTemplate.create(event.params.position)
```

Comment:
It's been a long time since I've coded a subgraph for the last time, and I didn't remember using template, could you please explain what it does?

Reply:
Subgraph index data from datasource but when we have a factory or deploy a contract on the fly and then we want to track it in subgraph as datasource that's where template is used.